src/js/button.js

```diff
@@ -7,13 +12,13 @@
  * @class
  * @constructor
  */
-vjs.Button = vjs.Component.extend({
+var Button = Component.extend({
   /**
    * @constructor
    * @inheritDoc
    */
   init: function(player, options){
```

Comment:
If you do change to es6 classes it will force developers to do bigger changes in their existing components/plugins besides the api changes. 

Don't get me wrong I am not saying that it should not be done. I am just wondering how much effort will it take to make everything work again build, test, demos, and so on.